### DISCUSSION TITLE
how to configure JWT signer for service that support multiple consumers

### DISCUSSION THREAD

**Discussion by smaddula862** [Kong/kong#13977] [Help]
i configured a JWT signer plugin in a service, same service would be used by multiple consumers.
if i need to validate channel token i need to use two different Channel JWKS.
how should i handle this use case ?

**Comment by Water-Melon** [comment]
Unfortunately, `jwt-signer` does not support consumers.
Hi @catbro666 , do you have any ideas on this?

**Comment by catbro666** [comment]
It doesn't matter how many consumers are using the service. The tokens of different consumers are normally issued by the same idp, and the jwks fetched from `channel_token_jwks_uri` will be used to verify the channel tokens.

**Reply by smaddula862** [reply]
in my case i have multiple consumers for the same API, we need to use different RSA public keys using JWKS URL for channel token validation  based on consumer.

**Reply by catbro666** [reply]
@smaddula862  jwt-signer plugin doesn't require any additional configuration to support this scenario. Different jwk's should have different `kid` and/or `x5t`, which will be included in the header of the issued token and then used by the jwt-signer to identify the key.
